src/builder.rs

```diff
@@ -347,6 +347,28 @@ impl<'req, 'dev, 'win, W: HasRawWindowHandle> PixelsBuilder<'req, 'dev, 'win, W>
     }
 }
 
+/// Compare the given size to the limits defined by `device`.
+///
+/// # Errors
+///
+/// - [`TextureError::TextureWidth`] when `width` is 0 or greater than GPU texture limits.
+/// - [`TextureError::TextureHeight`] when `height` is 0 or greater than GPU texture limits.
+pub fn check_texture_size(
+    device: &wgpu::Device,
+    width: u32,
+    height: u32,
+) -> Result<(), TextureError> {
+    let limits = device.limits();
```

Comment:
It will probably be better to store the limits in the constructor instead of making this call each time. I don't know what the exact overhead is, but we only need one of many limits here, and the limit is static.